[extractor/opencast] Add ltitools interface to OpencastPlaylistIE and fix Tests

C0D3D3V opened the pull request:
### Description

I'm the original author of the Opencast Extractor (bwildenhain then took care of it further so that it would be merged into yt-dlp. Thanks again for that), and I want to make an update to the extractor.  Opencast now has a new "interface" to access playlists, and that is now used more often than the old one. For that reason I extended the valid URLs of the OpencastPlaylist Extractor

Also I fixed the Tests. oc-video.ruhr-uni-bochum.de seems to be down currently so I replaced it with an other example. I also added the missing keys to the first test.

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

Grub4K reviewed (commented):
I would not remove the old test. If it is failing then mark it as `skip` with an appropriate message.

C0D3D3V commented:
@Grub4K there is not even a DNS record for oc-video.ruhr-uni-bochum.de anymore I am not sure if it will ever come back online. We could also just remove it from the hosts list.

Grub4K commented:
I am always for removing stuff but tests are usually kept for reference. In the past @pukkandan wanted to keep stuff around. I think its good to go since its always in the git history and removing this wont hurt since we can know its unused and we have equivalent tests. We'll see

pukkandan commented:
When new test covers the same URL pattern and code path, it's fine to remove old one

C0D3D3V commented:
I replaced `https://oc-video.ruhr-uni-bochum.de/engage/ui/index.html?e=1&p=1&epFrom=b1a54262-3684-403f-9731-8e77c3766f9a`
with `https://electures.uni-muenster.de/engage/ui/index.html?e=1&p=1&epFrom=39391d10-a711-4d23-b21d-afd2ed7d758c` 
so the same pattern is still covered.

And for the new pattern I added `https://oc-video1.ruhr-uni-bochum.de/ltitools/index.html?subtool=series&series=cf68a4a1-36b1-4a53-a6ba-61af5705a0d0&lng=de`. That is the same playlist as in the first playlist test, but a diffrent pattern. 

So the code coverage is the same as before (+ the new pattern). 

It could only be that the old playlist I choosed had videos with diffrent formats than the new one, so that a little bit more code was covered. But I remember that the tests I added back then did not cover all format paths. 
However expanding the code coverage was not meant to be part of this PR. But if it is requested I can add it or make a second PR in future.

C0D3D3V commented:
So please give me instructions what I should do that you are fine with it.

pukkandan commented:
Pls be patient. Will review it when I can

pukkandan commented on yt_dlp/extractor/opencast.py:
```diff
@@ -135,8 +138,10 @@ def _real_extract(self, url):
 class OpencastPlaylistIE(OpencastBaseIE):
     _VALID_URL = r'''(?x)
-                            https?://(?P<host>%s)/engage/ui/index.html\?.*?
-                            epFrom=(?P<id>%s)
+                            https?://(?P<host>%s)
+                            (/engage/ui/index.html\?.*?epFrom=|
+                            /ltitools/index.html\?.*?series=)
+                            (?P<id>%s)
                     ''' % (OpencastBaseIE._INSTANCES_RE, OpencastBaseIE._UUID_RE)
```
```suggestion
    _VALID_URL = rf'''(?x)
        https?://(?P<host>{OpencastBaseIE._INSTANCES_RE})(
            /engage/ui/index.html\?(?:[^#]+&)?epFrom=|
            /lttools/index.html\?(?:[^#]+&)?series=
        )(?P<id>{OpencastBaseIE._UUID_RE})'''
```

C0D3D3V replied:
Thank you very much, I did not use rf strings before because vs-code does not highlight the regex colloring on such stings. Actually I even have opened an issue because of this https://github.com/microsoft/pylance-release/issues/2757 :D But, if you prefer these I'm fine with it.

C0D3D3V commented:
Also sorry for the delay

pukkandan commented on yt_dlp/extractor/opencast.py:
```diff
@@ -105,10 +105,10 @@ def _parse_mediapackage(self, video):
-                    https?://(?P<host>%s)/paella/ui/watch.html\?.*?
-                    id=(?P<id>%s)
-                    ''' % (OpencastBaseIE._INSTANCES_RE, OpencastBaseIE._UUID_RE)
+    _VALID_URL = rf'''(?x)
+                    https?://(?P<host>{OpencastBaseIE._INSTANCES_RE})/paella/ui/watch.html\?(?:[^#]+&)?
+                    id=(?P<id>{OpencastBaseIE._UUID_RE})
+                    '''
```
```suggestion
    _VALID_URL = rf'''(?x)
        https?://(?P<host>{OpencastBaseIE._INSTANCES_RE})/paella/ui/watch.html\?
        (?:[^#]+&)?id=(?P<id>{OpencastBaseIE._UUID_RE})'''
```

pukkandan commented on yt_dlp/extractor/opencast.py:
```diff
@@ -134,10 +137,12 @@ def _real_extract(self, url):
-                            https?://(?P<host>%s)/engage/ui/index.html\?.*?
-                            epFrom=(?P<id>%s)
-                    ''' % (OpencastBaseIE._INSTANCES_RE, OpencastBaseIE._UUID_RE)
+    _VALID_URL = rf'''(?x)
+                            https?://(?P<host>{OpencastBaseIE._INSTANCES_RE})
+                            (/engage/ui/index.html\?(?:[^#]+&)?epFrom=|
+                            /ltitools/index.html\?(?:[^#]+&)?series=)
+                            (?P<id>{OpencastBaseIE._UUID_RE})
+                    '''
```
```suggestion
    _VALID_URL = rf'''(?x)
        https?://(?P<host>{OpencastBaseIE._INSTANCES_RE})(
            /engage/ui/index.html\?(?:[^#]+&)?epFrom=|
            /lttools/index.html\?(?:[^#]+&)?series=
        )(?P<id>{OpencastBaseIE._UUID_RE})'''
```

C0D3D3V replied:
I do not think this will work. That is the same change as you suggested before. I think the regex struggles with the newline

C0D3D3V replied:
yep I tested it, the tests fail

pukkandan replied:
just a typo
